### PR TITLE
Improvements to shiny and plot functions (solving several open issues)

### PR DESCRIPTION
**Pull request by warrenmcg**
Here is a list of issues solved and enhancements:
#124: this was due to a mistake in how shiny and the plots handled the new sleuth object API when its `obj$gene_mode` was `TRUE`. This is now fixed:
+ The error reporting is clearer when the user did not use `extra_bootstrap_summary` — and `read_bootstrap_tpm` if they want to see TPM) — which are essential for use with `transcript view` and `gene view`.
+ There was a bug which caused the `bs_quants` to still have `est_counts` as the label when the unit was `scaled_reads_per_base`. This is likely what was causing the problem for users who used the options above and still had an error. This is now fixed.
+ These changes also addressed #126, since now the plot functions make sure that the right unit is used depending on whether the sleuth object is in `transcript mode` or `gene mode`.

In addition to the above changes, several changes were made to the `transcript view` and `gene view` to accommodate when `gene_mode` is `TRUE`:
+ When `obj$gene_mode` is `TRUE`, the `gene view` on shiny now acts similarly to the old `transcript view`, where you can specify a gene ID and you'll get the bootstrap plot for that gene. It correctly can handle either `TPM` or `scaled_reads_per_base`. Similar to the `gene view` for transcripts, you can use alternative gene identifiers.
+ Also, when `obj$gene_mode` is `TRUE`, `transcript view` is now disabled (since there is no way to access normalized transcript data without rerunning `sleuth prep`). I couldn't figure out how to easily hide the tab, so the tab just shows a message.

In addition to #124, there were other issues handled in this pull request:
+ fixed the bug found in #82 for `plot_qq`. Thanks to @a-slide for identifying the problem!
+ to solve #66, enhanced `plot_group_density` to handle more than one covariate, and improved the error reporting for that function.
+ improved the sample heat map function to address #67 and #87. This also led to improving the transcript heatmap function, which could not handle the `scaled_reads_per_base` unit if the object was in `gene mode`:
  - Both use the `pheatmap` package now
  - Both can annotate covariates from `obj$samples_to_covariates`
  - cluster samples in sample heat map and transcripts in transcript heat map
  - the plot functions now can take additional arguments to further customize the heat maps when not done in `sleuth_live`.

Remaining to-dos:
- [x] ~~enhance shiny to handle the extra heatmap functionality.~~

### CODE REVIEW

**Comment by warrenmcg** [comment]
Additional note: I added in the `sleuth_live` documentation from the `documentation` branch.

File: R/plots.R [hunk 1]
```diff
@@ -318,7 +318,7 @@ plot_pc_variance <- function(obj,
 #' @param units either 'est_counts' ('scaled_reads_per_base' for gene_mode) or 'tpm'
 #' @param trans a string pointing to a function to use for the transformation.
 #' @param grouping a string from the columns of \code{sample_to_covariates} in
-#' the sleuth object for which to group and color by
+#' the sleuth object for which to group and color by. At this time,
```
Comment (pimentel) [inline R/plots.R]: do you remember what you were going to type here?

File: R/sleuth.R [hunk 2]
```diff
@@ -858,6 +864,46 @@ transcripts_from_gene <- function(obj, test, test_type,
   table$target_id[table[, 2] == gene_name]
 }
 
+#' Get the gene ID using other gene identifiers
+#'
+#' Get the \code{target_id} of a gene using other gene identifiers
+#'
+#' @param obj a \code{sleuth} object
+#' @param test a character string denoting which beta to use
```
Comment (pimentel) [inline R/sleuth.R]: note to self: remove extra documentation here

File: R/plots.R (hunk 1 above)
Comment (warrenmcg) [reply R/plots.R]: Oh, whoops. This comment was originally going to say that "At this time, only one column can be used"; however, I implemented a version that can handle more than one column, as they exist in the table. You can revert this line back.

**Pull request merged**